src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs

```diff
@@ -308,7 +308,7 @@ public static MediaTypeSegmentWithQuality CreateMediaTypeSegmentWithQuality(stri
 
             // We check if the parsed media type has value at this stage when we have iterated
             // over all the parameters and we know if the parsing was sucessful.
-            if (!parser.ParsingFailed)
+            if (!parser.ParsingFailed && parser.CurrentOffset >= start)
```

Comment:
This comment needs to be updated to reflect **why** `parser.CurrentOffset` is significant. Even better, add a property to the parser with a name that reflects the actual important semantic